client/task-list/tasks/payments/RemotePayments/components/PaymentConnect.js

```diff
@@ -28,53 +28,50 @@ export const PaymentConnect = ( {
 		title,
 	} = paymentGateway;
 
-	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
 	const { createNotice } = useDispatch( 'core/notices' );
-	const slot = useSlot( `woocommerce_remote_payment_form_${ key }` );
+	const { updatePaymentGateway } = useDispatch( PAYMENT_GATEWAYS_STORE_NAME );
+	const slot = useSlot( `woocommerce_remote_payment_form_${ id }` );
 	const hasFills = Boolean( slot?.fills?.length );
 	const fields = settingKeys
 		? settingKeys.map( ( settingKey ) => settings[ settingKey ] )
 		: [];
 
-	const isOptionsRequesting = useSelect( ( select ) => {
-		const { isOptionsUpdating } = select( OPTIONS_STORE_NAME );
+	const { isUpdating } = useSelect( ( select ) => {
+		const { isPaymentGatewayUpdating } = select(
+			PAYMENT_GATEWAYS_STORE_NAME
+		);
 
-		return isOptionsUpdating();
+		return {
+			isUpdating: isPaymentGatewayUpdating(),
+		};
 	} );
 
-	const updateSettings = async ( values ) => {
-		recordConnectStartEvent( key );
-
-		const options = {};
-
-		fields.forEach( ( field ) => {
-			const optionName = field.option || field.name;
-			options[ optionName ] = values[ field.name ];
-		} );
-
-		if ( ! Object.keys( options ).length ) {
-			return;
-		}
-
-		const update = await updateOptions( {
-			...options,
-		} );
-
-		if ( update.success ) {
-			markConfigured( key );
-			createNotice(
-				'success',
-				title + __( ' connected successfully', 'woocommerce-admin' )
-			);
-		} else {
-			createNotice(
-				'error',
-				__(
-					'There was a problem saving your payment settings',
-					'woocommerce-admin'
-				)
-			);
-		}
+	const handleSubmit = async ( values ) => {
+		recordConnectStartEvent( id );
+
+		updatePaymentGateway( id, {
```

Comment:
Do we want there to be an `await` here?

Reply:
Since this is an async function, we don't need await here.  I can remove the above `async` on this function though and things should continue to work as expected.

Reply:
Yeah, my thought as well. As written this just returns a promise that resolves to _undefined_ I believe, so I wasn't sure of the intention. It doesn't look like this callback is utilized as a promise at any point so removing the `async` would more clearly indicate that this is just a `=> void` function. I'd consider it option, but was curious.